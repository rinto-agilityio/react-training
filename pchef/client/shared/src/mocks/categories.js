export const categories = [
  {
    id: 'chicken',
    name: 'Chicken',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Heathy',
    name: 'Heathy',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Breakast',
    name: 'Easy Breakast',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Dessert',
    name: 'Dessert',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Fish',
    name: 'Fish',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Pizza',
    name: 'Pizza',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Cake',
    name: 'Cake',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Deserts',
    name: 'Deserts',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Salad',
    name: 'Salad',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Kids',
    name: 'Kids',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Meat',
    name: 'Meat',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Lunch',
    name: null,
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
]
