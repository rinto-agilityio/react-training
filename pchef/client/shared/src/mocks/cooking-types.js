export const cookingTypes = [
  {
    id: 'baking',
    name: 'Baking',
  },
  {
    id: 'roasting',
    name: 'Roasting',
  },
  {
    id: 'frying',
    name: 'Frying',
  },
]
