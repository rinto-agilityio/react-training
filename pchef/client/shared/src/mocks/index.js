// mock categories
export const category = {
  name: 'Heathy',
  imgUrl:
    'https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
}
