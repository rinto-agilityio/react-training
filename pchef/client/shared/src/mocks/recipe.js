export const recipe = {
  description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
  title: 'Bun Bo Hue Recipe',
  imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
  votes: [1],
  steps: [
    {
      step: 1,
      description: 'Step 1 description',
      title: 'Step 1 title',
      subTitle: 'Step 1 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 2,
      description: 'Step 2 description',
      title: 'Step 2 title',
      subTitle: 'Step 2 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 3,
      description: 'Step 3 description',
      title: 'Step 3 title',
      subTitle: 'Step 3 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 4,
      description: 'Step 4 description',
      title: 'Step 4 title',
      subTitle: 'Step 4 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 5,
      description: 'Step 5 description',
      title: 'Step 5 title',
      subTitle: 'Step 5 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 6,
      description: 'Step 6 description',
      title: 'Step 6 title',
      subTitle: 'Step 6 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
  ],
}
