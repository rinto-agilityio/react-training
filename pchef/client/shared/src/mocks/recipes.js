export const recipes = [
  {
    id: 0,
    title: 'Bun Bo Hue Recipe',
    subTitle: 'Traditional food',
    description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
    imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    votes: [1],
    views: 1,
    steps: [
      {
        step: 1,
        description: 'Step 1 description',
        title: 'Step 1 title',
        subTitle: 'Step 1 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 2,
        description: 'Step 2 description',
        title: 'Step 2 title',
        subTitle: 'Step 2 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 3,
        description: 'Step 3 description',
        title: 'Step 3 title',
        subTitle: 'Step 3 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 4,
        description: 'Step 4 description',
        title: 'Step 4 title',
        subTitle: 'Step 4 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 5,
        description: 'Step 5 description',
        title: 'Step 5 title',
        subTitle: 'Step 5 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 6,
        description: 'Step 6 description',
        title: 'Step 6 title',
        subTitle: 'Step 6 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
    ],
  },
  {
    id: 1,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 2,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 3,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
]
