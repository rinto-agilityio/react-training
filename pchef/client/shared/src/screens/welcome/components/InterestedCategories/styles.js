import { StyleSheet } from 'react-native'
import { METRICS, COLORS, FONTS } from '../../../../themes'

const styles = StyleSheet.create({
  item: {
    margin: 1,
    position: 'relative',
  },
  layer: {
    position: 'absolute',
    backgroundColor: COLORS.black,
    zIndex: 1,
  },
  activeLayer: {
    opacity: METRICS.activeItem,
  },
  inactiveLayer: {
    opacity: METRICS.inactiveItem,
  },
  name: {
    position: 'absolute',
    fontWeight: 'bold',
    zIndex: 2,
  },
  activeName: {
    color: COLORS.white,
  },
  inactiveName: {
    color: COLORS.welcome.disableItem,
  },
  primaryName: {
    left: METRICS.smallMargin,
    top: 8,
    fontSize: FONTS.fontSize.base,
  },
  secondaryName: {
    left: METRICS.mediumMargin,
    top: METRICS.smallMargin + METRICS.mediumMargin,
    fontSize: FONTS.fontSize.large,
  },
})

export default styles
