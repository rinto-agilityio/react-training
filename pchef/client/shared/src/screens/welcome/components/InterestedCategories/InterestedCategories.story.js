// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { View } from 'react-native'
import { action } from '@storybook/addon-actions'
import { categories } from '../../../../mocks'

// Components
import InterestedCategories from '.'

// define props
const props = {
  onChooseCategory: action('clicked'),
  activeList: [categories[0].id, categories[1].id],
  categories,
}

storiesOf('InterestedCategories', module)
  .add('primary', () => (
    <View style={{ width: 400 }}>
      <InterestedCategories {...props} />
    </View>
  ))
  .add('secondary', () => (
    <View style={{ width: 800 }}>
      <InterestedCategories {...props} type="secondary" />
    </View>
  ))
