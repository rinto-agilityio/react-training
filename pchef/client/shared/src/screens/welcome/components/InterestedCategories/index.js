import React, { useState } from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import Wrapper from '../../../../layout/Wrapper'
import styles from './styles'

type Props = {
  type?: string,
  onChooseCategory: (id: string) => Promise<void>,
  activeList: Array<string>,
  categories: Array<{ id: string, name: string, imgUrl: string }>,
}

const InterestedCategories = ({
  type = 'primary',
  activeList,
  onChooseCategory,
  categories = [],
}: Props) => {
  // declare state item size
  const [itemSize, setItemSize] = useState({ width: 0, height: 0 })

  // get wrapper size
  const handlingGetLayout = (event = { nativeEvent: { layout: {} } }) => {
    const { width } = event.nativeEvent.layout

    // get item Size, three items in a row, margin between items is 2px
    const itemWidth = (width - 6) / 3

    // set item size
    setItemSize({
      width: itemWidth,
      height: itemWidth,
    })
  }

  return (
    <Wrapper
      direction="row"
      childPosition="right"
      customStyles={styles.container}
      onLayout={handlingGetLayout}
    >
      {categories.map(category => {
        // check if category has been chosen
        const chosen = activeList.includes(category.id)

        return (
          <View key={category.id} style={[styles.item, itemSize]}>
            <TouchableOpacity onPress={() => onChooseCategory(category.id)}>
              <Image
                style={[styles.image, itemSize]}
                source={{ uri: category.imgUrl }}
              />
              <Text
                style={[
                  styles.name,
                  styles[`${type}Name`],
                  styles[`${chosen ? 'active' : 'inactive'}Name`],
                ]}
              >
                {category.name}
              </Text>
              <View
                style={[
                  styles.layer,
                  itemSize,
                  styles[`${chosen ? 'active' : 'inactive'}Layer`],
                ]}
              />
            </TouchableOpacity>
          </View>
        )
      })}
    </Wrapper>
  )
}

InterestedCategories.defaultProps = {
  type: 'primary',
}

export default InterestedCategories
