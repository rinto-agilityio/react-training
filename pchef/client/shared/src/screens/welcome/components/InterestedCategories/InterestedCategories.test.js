// Components
import InterestedCategories from '.'
import { categories } from '../../../../mocks'

it('InterestedCategories snapshots', () => {
  // define props
  const props = {
    onChooseCategory: () => {},
    activeList: [categories[0].id, categories[1].id],
  }

  const primaryComponent = shallow(
    <InterestedCategories {...props} type="primary" customStyle={{ flex: 1 }} />,
  )
  const secondaryComponent = shallow(
    <InterestedCategories {...props} type="secondary" />,
  )

  expect(primaryComponent).toMatchSnapshot()
  expect(secondaryComponent).toMatchSnapshot()
})

it('InterestedCategories actions', () => {
  // define props
  const props = {
    customStyle: {},
    onChooseCategory: jest.fn(),
    activeList: [categories[0].id, categories[1].id],
  }

  const component = shallow(<InterestedCategories {...props} />)

  // simulate press item
  component.props().onLayout()
  component
    .find('TouchableOpacity')
    .at(0)
    .props()
    .onPress()

  expect(props.onChooseCategory).toHaveBeenCalled()
})
