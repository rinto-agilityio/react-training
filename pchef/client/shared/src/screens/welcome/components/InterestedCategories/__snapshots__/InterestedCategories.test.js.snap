// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestedCategories snapshots 1`] = `ShallowWrapper {}`;

exports[`InterestedCategories snapshots 2`] = `ShallowWrapper {}`;
